subprojects/messaging/src/main/java/org/gradle/internal/serialize/DefaultSerializerRegistry.java

```diff
@@ -49,42 +52,62 @@ public DefaultSerializerRegistry(boolean supportClassHierarchy) {
 
     @Override
     public <T> void register(Class<T> implementationType, Serializer<T> serializer) {
-        serializerMap.put(implementationType, serializer);
+        readWriteLock.writeLock().lock();
```

Comment:
I feels wrong to add locks to services. Something different is wrong here. Why are serializers registered at runtime in parallel? That should not happen.

Reply:
Who would be the right person to ask this question to? I personally don't have enough context regarding this code to answer this question.

Reply:
If you can track down who wrote the `readState` and `writeState` methods in IsolatableSerializerRegistry to ask them why `useJavaSerialization` is called, that might help.

It's surprising to me that encoding and decoding using the serializer _registry_ changes the state of the registry.